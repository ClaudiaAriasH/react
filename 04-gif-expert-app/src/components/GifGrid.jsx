import { getGifs } from "../helpers/getGifs";

export const GifGrid = ({ category }) => {
  getGifs(category);

  return (
    <>
      <div key={category}>
        <li>{category}</li>
      </div>
    </>
  );
};
