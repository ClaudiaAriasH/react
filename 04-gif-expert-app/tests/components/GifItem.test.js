import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe("Prueba <GifItem/>", () => {
  const title = "One Punch";
  const url = "https://giphy.com/embed/VXJWhaO7afRe";

  test("debe de hacer match con el snapshot", () => {
    const { container } = render(<GifItem title={title} url={url} />);
    expect(container).toMatchSnapshot();
  });

  test("debe de tener la imagen igual al url y alt", () => {
    const { container } = render(<GifItem title={title} url={url} />);

    const { src, alt } = screen.getByRole("img");

    expect(src).toBe(url);
    expect(alt).toBe(title);
  });

  test("debe de mostrar el titulo del componente", () => {
    const { container } = render(<GifItem title={title} url={url} />);

    expect(screen.getByText(title)).toBeTruthy;
    
  });
});
