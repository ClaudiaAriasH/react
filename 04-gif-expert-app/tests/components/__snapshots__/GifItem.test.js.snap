// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba <GifItem/> debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="One Punch"
      src="https://giphy.com/embed/VXJWhaO7afRe"
    />
    <p>
      One Punch
    </p>
  </div>
</div>
`;
